### ISSUE TITLE
FR: merge technical replica's scRNAseq pipeline

### ISSUE THREAD

**Issue by JGASmits** [vanheeringen-lab/seq2science#736]
It would be nice to be able to merge fastqs of technical replica's in the scRNAseq pipeline similar to how the bulk pipelines support this. Right now this isnt working yet, if i inc. a column with 'technical replica'  it complains that the values in it should be unique.
No hurry though, CAT on the fastqs and inputting those in s2s seems to work fine!

**Comment by tilschaef** [comment]
I relabeld this one since I get the impression that it is supposed to work.  Adding to the issue @JGASmits  described above, I tested the merging with the samples.tsv below

```
sample	assembly	technical_replicates
CMdiff10	GRCh38.p13	CMdiff
CMdiff12	GRCh38.p13	CMdiff
```
The fastq files are stored locally as CMdiff10_{R1,R2}.fastq and CMdiff12_{R1,R2}.fastq.gz.

However, this does not seem to work and s2s exits with `Missing input files for rule rename_sample`

**Comment by Maarten-vd-Sande** [comment]
The problem is that the input to kallistobus is: `rules.fastq_pair.output.reads`, and that the rule `fastq_pair` uses the trimmed fastq (as it should). However, technical replicates get merged after trimming (performance reasons, but also in the edge-case someone used different adapters). I made an overview of the problem, the red links are missing:

![image](https://user-images.githubusercontent.com/15323399/128007639-5d381202-4e5d-49e9-9e1b-cb5414888110.png)

This is specific to scRNA-seq because it is the only workflow using either the `fastq-pair` rule or the `kallistobus` rule. I can pick this up if you want @tilschaef or you can have a go at it yourself if you want :smile:

**Comment by tilschaef** [comment]
Thanks @Maarten-vd-Sande for the explanation and illustration :). I'll have a look at it.

**Comment by Maarten-vd-Sande** [comment]
Not sure if you are familiar with the concept of ruleorder yet, but that is definitely sth you'd need together with some if statement(s): https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#handling-ambiguous-rules

**Comment by tilschaef** [comment]
Implemented in #739
